src/main/java/com/walmartlabs/x12/rule/TrailerSegmentCountX12Rule.java

```diff
@@ -0,0 +1,108 @@
+package com.walmartlabs.x12.rule;
+
+import com.walmartlabs.x12.SegmentIterator;
+import com.walmartlabs.x12.X12Segment;
+import com.walmartlabs.x12.X12TransactionSet;
+import com.walmartlabs.x12.exceptions.X12ErrorDetail;
+import com.walmartlabs.x12.exceptions.X12ParserException;
+import com.walmartlabs.x12.standard.StandardX12Parser;
+import com.walmartlabs.x12.util.ConversionUtil;
+
+import java.util.List;
+
+public class TrailerSegmentCountX12Rule implements X12Rule {
+
+    /**
+     * check each trailer record and verify 
+     * that the total segments match what 
+     * is reported in the trailer count
+     * 
+     * @param segmentList
+     */
+    @Override
+    public void verify(List<X12Segment> segmentList) {
+        
+        SegmentIterator segments = new SegmentIterator(segmentList);
+
+        int groupHeaders = 0;
+        int groupTrailers = 0;
+        int transactionCount = 0;
+        String currentGroupControlNumber = null;
+        int groupCountOnIeaTrailer = this.findGroupCountOnIeaTrailer(segmentList);
+        
+        while (segments.hasNext()) {
+            X12Segment currentSegment = segments.next();
+            
+            if (StandardX12Parser.GROUP_HEADER_ID.equals(currentSegment.getIdentifier())) {
+                groupHeaders++;
+                currentGroupControlNumber = currentSegment.getElement(6);
+            }
+            
+            if (X12TransactionSet.TRANSACTION_SET_HEADER.equals(currentSegment.getIdentifier())) {
+                transactionCount++;
+            }
+            
+            if (StandardX12Parser.GROUP_TRAILER_ID.equals(currentSegment.getIdentifier())) {
+                groupTrailers++;
+                this.verifyTransactionsOnGroupTrailer(currentGroupControlNumber, transactionCount, currentSegment);
+                
+                // reset transaction numbers
+                transactionCount = 0;
+                currentGroupControlNumber = null;
+            }
+        }
+        
+        this.verifyInterchangeControlTrailer(groupHeaders, groupTrailers, groupCountOnIeaTrailer);
+    }
+    
+    /**
+     * find IEA segment and get number of groups
+     */
+    private int findGroupCountOnIeaTrailer(List<X12Segment> segmentList) {
+        int groupCountOnIeaTrailer = 0;
+        int segmentCount = segmentList.size();
+        
+        X12Segment ieaTrailer = segmentList.get(segmentCount - 1);
+        if (StandardX12Parser.ISA_TRAILER_ID.equals(ieaTrailer.getIdentifier())) {
+            groupCountOnIeaTrailer = ConversionUtil.convertStringToInteger(ieaTrailer.getElement(1));
```

Comment:
`ConversionUtil.convertStringToInteger();` returns `Integer` and `groupCountOnIeaTrailer` is int. could result in Exception.